File: lib/models/subject.dart
// ignore_for_file: non_constant_identifier_names

class Subject {
  String? subject_id;
  String? subject_name;
  String? subject_description;
  String? subject_price;
  String? subject_sessions;
  String? subject_rating;

  Subject({
    this.subject_id,
    this.subject_name,
    this.subject_description,
    this.subject_price,
    this.subject_sessions,
    this.subject_rating,
  });

  Subject.fromJson(Map<String, dynamic> json) {
    subject_id = json['subject_id'].toString();
    subject_name = json['subject_name'];
    subject_description = json['subject_description'];
    subject_price = json['subject_price'].toString();
    subject_sessions = json['subject_sessions'].toString();
    subject_rating = json['subject_rating'].toString();
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['subject_id'] = subject_id;
    data['subject_name'] = subject_name;
    data['subject_description'] = subject_description;
    data['subject_price'] = subject_price;
    data['subject_sessions'] = subject_sessions;
    data['subject_rating'] = subject_rating;
    return data;
  }
}

File: php/subject.php
<?php
include("dbconnect.php");
if (!isset($_POST)) {
    $response = array('status' => 'failed', 'data' => null);
    sendJsonResponse($response);
    die();
}

$limit = 5;
$page = (isset($_POST['page']) && is_numeric($_POST['page'])) ? $_POST['page'] : 1;
$pageStart = ($page - 1) * $limit;
$sqlselectSubject = "SELECT * FROM tbl_subjects LIMIT $pageStart, $limit";
$stmt = $conn->prepare($sqlselectSubject);
$stmt->execute();
$result = $stmt->get_result();

$sql = $conn->query("SELECT count(subject_id) AS id FROM tbl_subjects")->fetch_assoc();
$allRecords = $sql['id'];
$totalPages = ceil($allRecords / $limit);
$prev = $page - 1;
$next = $page + 1;

if ($allRecords > 0) {
    $subjects['subjects'] = array();
    while ($row = $result->fetch_assoc()) {
        $sublist = array();
        $sublist['subject_id'] = $row['subject_id'];
        $sublist['subject_name'] = $row['subject_name'];
        $sublist['subject_description'] = $row['subject_description'];
        $sublist['subject_price'] = $row['subject_price'];
        $sublist['subject_sessions'] = $row['subject_sessions'];
        $sublist['subject_rating'] = $row['subject_rating'];
        array_push($subjects['subjects'], $sublist);
    }
    $response = array('status' => 'success', 'page' => "$page", 'totalPages' => "$totalPages", 'data' => $subjects);
    sendJsonResponse($response);
} else {
    $response = array('status' => 'failed', 'page' => "$page", 'totalPages' => "$totalPages", 'data' => null);
    sendJsonResponse($response);
}


function sendJsonResponse($sentArray)
{
    header('Content-Type: application/json');
    echo json_encode($sentArray);
}
?>